### DISCUSSION TITLE
استفاده از CDN در مرزبان

### DISCUSSION THREAD

**Discussion by JavadWolf-af** [Gozargah/Marzban#403] [Ideas]
سلام دوستان برای استفاده CDN در مرزبان باید طبق اموزش زیر جلو برید.. (من برای کلادفلر مثال میزنم برای ابرآروان هم بهمین شکله)

1.مرحله اول باید یه ساب تعریف کنید و تیک پروکسی رو روشن بزارید..
![Annotation 2023-07-24 234423](https://github.com/Gozargah/Marzban/assets/22157772/2bfe5906-7140-4eec-98e6-c01ce494921a)

2.مرحله دوم باید به پنل مرزبان برید و کد زیر رو به CORE SETTING اضافه کنید..

نکات مهم : 
*حتما پورت 443 باشد چون برای کارکردن روی CDN واجب هست...
*حتما ادرس سرتیفیکیت خودتان را وارد کنید..

    {
      "tag": "VLESS_INBOUND",
      "listen": "0.0.0.0",
      "port": 443,
      "protocol": "vless",
      "settings": {
        "clients": [],
        "decryption": "none"
      },
      "streamSettings": {
        "network": "ws",
        "wsSettings": {
          "path": "/"
        },
        "security": "tls",
        "tlsSettings": {
          "serverName": "آدرس ساب دامنه که تیکش را روشن کردید وارد کنید",
          "certificates": [
            {
              "certificateFile": "/var/lib/marzban/certs/fullchain.pem",
              "keyFile": "/var/lib/marzban/certs/key.pem"
            }
          ],
          "minVersion": "1.2",
          "cipherSuites": "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256:TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256:TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384:TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384:TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256:TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"
        }
      }
    },

3. در قسمت HOST SETTINGS این گزینه جدید میاد و روش کلیک میکنید..

![Annotation 2023-07-25 002201](https://github.com/Gozargah/Marzban/assets/22157772/a796a50f-944c-4e0f-920b-1d172c6bb6ee)

4. و وقتی کلیک کردید تمامی فیلد هارو مانند تصویر زیر پرکنید و APPLY را بزنید.

![Annotation 2023-07-25 002548](https://github.com/Gozargah/Marzban/assets/22157772/d3e43775-0e8a-41ab-a1f7-eea4004d4745)

5.اپدیت کنید و متصل شوید...

نکته:
*برخی دامنه ها جواب نمیدهند مانند .SITE , .TECH من برای ابر آروان روی .IR تست کردم جواب داد..

موفق باشید...

کوچیک شما Phoenix

**Comment by totem2** [comment]
سلام دمت گرم داداش یه کانفیگ دیگه بود grpc که جای آدرس آیپی تمیز کلود میزدیم sni هم ساب دامین اونم میتونی کدشو بزاری؟

**Reply by shahinmagic1** [reply]
فقط روی cdn gcore تست میکنم جواب نمیده میشه راهنمایی کنید ؟

**Reply by JavadWolf-af** [reply]
سلام خودمم روی g-core تست کردم ظاهرن جواب نمیده

**Comment by JavadWolf-af** [comment]
{
            "tag": "VLESS GRPC TLS",
            "listen": "0.0.0.0",
            "port": 2049,
            "protocol": "vless",
            "settings": {
                "clients": [],
                "decryption": "none"
            },
            "streamSettings": {
                "network": "grpc",
                "grpcSettings": {
                    "serviceName": "abc"
                },
                "security": "tls",
                "tlsSettings": {
                    "serverName": "SERVER_NAME",
                    "certificates": [
                        {
                            "ocspStapling": 3600,
                            "certificateFile": "/var/lib/marzban/certs/fullchain.pem",
                            "keyFile": "/var/lib/marzban/certs/key.pem"
                        }
                    ],
                    "minVersion": "1.2",
                    "cipherSuites": "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256:TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256:TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384:TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384:TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256:TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"
                }
            }
        },

**Reply by JavadWolf-af** [reply]
این بود منظورت؟

**Comment by YasharMir** [comment]
دوست عزیز اگر کانفیگ های زیر را هم داری بیزحمت بزار

VLESS-gRPC-REALITY
VLESS-WSS-Nginx
VLESS-TCP-TLS-WS
و اینکه نحوه راه اندازی ReverseProxy در مرزبان چگونه است؟

**Reply by JavadWolf-af** [reply]
سلام اگه بازم خاستی بگو برات بزارم 
ریورس پروکسی روی مرزبان نمیشه چون باید تو دیوایس تغییر بدی و مرزبان قابلیت اپدیت سابسکریپشن داره که با هربار اپدیت و تغییر مجدد تنظیم میشه

**Comment by zapzocket** [comment]
سلام عزیز یه کمکی بتونی بکنی ممنون میشم من با دامنه ssl دار وارد پنل میشم وقتی تیک پروکسی داخل اروان رو روشن میکنم دیگه پنل باز نمیشه راه حلش چیه؟

**Reply by JavadWolf-af** [reply]
سلام باید پورت دامنه اتو به 443 تغییر بدی اروان فقط روی این پورت میتونه پروکسی روشن جواب بده

**Reply by JavadWolf-af** [reply]
اونی که برای پنلته رو تیکشو خاموش کن تا بتونی از پورت 443 تو کانفیگت استفاده کنی

**Comment by Mjzaret** [comment]
برای سرورهای نود این روش جواب میده؟

**Reply by JavadWolf-af** [reply]
بله

**Reply by MILADALEF** [reply]
برای نود در قسمت سرتیفیکیت چه آدرسی بزنیم ؟

**Reply by JavadWolf-af** [reply]
نود ادرس سرتیفیکیت نمیخاد فقط کافیه ایپی نودت رو پشت یه نیم سرور جدید ببری و تیکشو روشن کنی و تو هاست ستینگ بزارین

**Comment by macbrave** [comment]
اگه آدرس پنل جدا باشه و یا چندین دامنه داشته باشیم چیکار باید بکنیم

**Reply by JavadWolf-af** [reply]
ربطی به ادرس پنل جدا نداره شما دامنه ست شده روی اروان یا کلادفلر رو بزنید.. که تیک پراکسی روشنه مرزبان قابلیت چندین پراکسی روشن و دی ان اس همزمان داره..

**Comment by vahidxo** [comment]
این موردی که گفتین رو من تست کردم و فقط روی سرور مستر جواب میده
برای پشت cdn بردن نودها چه کاری باید انجام بدیم؟

**Reply by JavadWolf-af** [reply]
فقط کافیه یه ساب دامین بسازی و ایپی نودت رو پشتش قرار بدی و تیک پراکسی رو روشن کنی و در قسمت adress توی هاست ستینگ بزاری..

**Reply by arshaalone** [reply]
> فقط کافیه یه ساب دامین بسازی و ایپی نودت رو پشتش قرار بدی و تیک پراکسی رو روشن کنی و در قسمت adress توی هاست ستینگ بزاری..
> 

خب اینجا یه موردی هست نود قبول نمیکنه وقتی دامین رو میزاریم فقط باید ایپی بذاریم که بعدش روی همراه فعال نیست

**Reply by alimarmooz** [reply]
سلام
دوست عزیز من سرت مرزبانم رو با unicorn گرفتم و امکانش هست اینباند cdn برای نود رو اینجا قرار بدین ؟
میخوام بدونم ادرس سرت رو چی بزارم
سی دی ان نود من اصلا کار نمیکنه !
چند روزه درگیرشم و نمیدونم دیگه چه کار کنم 🤦

**Comment by parham2013** [comment]


**Reply by JavadWolf-af** [reply]
سلام دوست عزیز اگه میخای نود رو پراکسی کنی ساب دامنه نود رو بزارو در قسمت هاست ستینگ دامنه نود رو بزار و اینکه سرتیفیکیت خود پنلو بزار نود سرتیفیکیت نمیخاد

**Comment by parham2013** [comment]


**Reply by JavadWolf-af** [reply]
این جدید بود خوبه که گفتی بچه ها استفاده میکنن

**Reply by Mjzaret** [reply]
من تمامی این موارد رو انجام دادم اما متاسفانه برای من کار نکرد
هم از سرتیفیکیت های کلودفلر استفاده کردم هم با سرتبات سرتیفیکت گرفتم هیچ کدومشون کار نکرد.
خود پنل کار میکنه و مشکلی نداره ولی نودها رو با روش بالا نتونستم وصل کنم.
هیچ پینگی نمیده.

**Reply by aqilahmadi** [reply]
سلام تونستین روشی برای این مشکل پیدا کنید چون منم الان همین مشکل رو دارم

**Comment by ben-aslan** [comment]
سلام با.ir هم کار میکنه این روش توی کلودفلر؟

**Reply by abedin70** [reply]
آره

**Comment by alimarmooz** [comment]
با سلام
من پنل رو با cady سرت گرفت و هیچ مشکلی با پنلم ندارم
ولی من میخوام با سرور نودم و با دامنه های مختلف نه ساب دامین مختلف !!
دامنه های مختلف از سی دی ان کلودفلیر استفاده کنم
با هیچ روشی نتونستم
کسی انجام داده و بتونه راهنماییم کنه !
نمونه اینباندی بزاره یا env رو برام بفرسته
شاید اونجاها مشکلی دارم

**Comment by MAsterhadi** [comment]
سلام من میخوام دوتا دامنه داشته باشم که یدونش برای بالا اومدن خود پنل باشه بدون  cdn با Tls   و یدونه برای کانفینگ ها که cdn روشن باشه  این دامنه دومو چطوری به پنل اضافه کنم ؟

**Comment by sirr8max** [comment]
سلام وقتت بخیر باشه، من میخوام کانفینگ VLESS+TCP+REALITY بسازم و داخل تنظیمات tcp میخوام گزینه TCP Fast Open فعال کنم میشه راهنمایی کنید ممنون میشم [Uploading Screenshot 2025-07-10 201024.png…]()
